### PR TITLE
*: add document for expensive query log

### PR DESCRIPTION
**Pull request by XuHuaiyu**
### What is the related PR or file link(s)? 

- [x] Reference link(s):
https://github.com/pingcap/tidb/pull/10799
https://github.com/pingcap/tidb/pull/10350
https://github.com/pingcap/tidb/pull/12809

- [ ] This PR is to align with:
- [ ] N/A (not applicable)

### Which TiDB version(s) does your changes apply to? 

- [ ] All active versions: dev, v3.0, v2.1, v3.1
- [x] dev (the latest development version)
- [x] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] All active and inactive versions
- [ ] N/A (not applicable)

**Note:** If your changes apply to multiple TiDB versions, make sure you update the documents in the **corresponding** version folders such as "dev", "v3.0", "v2.1" and "v3.1" in this PR.

- [x] Updated one version first. Will update other versions after I get two LGTMs.

### CODE REVIEW

File: dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+---
+title: 定位消耗系统资源多的查询
+category: how-to
+---
+
+# expensive query 日志
+
+TiDB 会将执行时间超过 tidb_expensive_query_time_threshold（默认值为 60s），或使用内存超过 [mem-quota-query](https://pingcap.com/docs-cn/stable/reference/configuration/tidb-server/configuration-file/#mem-quota-query)（默认值为 32 GB的语句输出到 [tidb-server 日志文件](https://pingcap.com/docs-cn/stable/reference/configuration/tidb-server/configuration-file/#logfile)（默认文件为：“tidb.log”）中，用于帮助用户在语句执行结束前定位由于执行时间过长或使用内存过多的语句，分析和解决语句执行的性能问题。
+
+注意，expensive query 日志和[慢查询日志](https://pingcap.com/docs-cn/dev/how-to/maintain/identify-slow-queries/)的区别是，慢查询日志是在语句执行完后才打印，expensive query 日志可以将正在执行中且超过资源限制的语句及其相关信息打印出来。
```
Comment (zz-jason) [inline dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md]: 可否多介绍一些实现原理和机制，供感兴趣的用户知其所以然？

Comment (XuHuaiyu) [reply dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md]: 简单改了下表述，`原理和机制` 在这里不好多写，这个文档中的文字，尽可能只和定位和解决问题相关会更好

File: dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+和 SQL 执行的用户相关的字段：
+
+* `user`：执行语句的用户名。
+* `conn_id`：用户的链接 ID，可以用类似 `con:60026` 的关键字在 TiDB 日志中查找该链接相关的其他日志。
+* `database`：执行语句时使用的 database。
+
+和 TiKV Coprocessor Task 相关的字段：
+
+* `wait_time`：该语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队；当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
```
Comment (qw4990) [inline dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md]: `该语句在 TiKV 的等待时间之和` -> `该语句所有 Coprocessor 请求在 TiKV 的等待时间之和`

**Comment by lilin90** [comment]
@yikeke Please take a language and format review. Thanks!

File: dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+* `cost_time`：日志打印时语句已经花费的执行时间。
+* `stats`：语句涉及到的表或索引使用的统计信息版本。值为 pesudo 时表示无可用统计信息，需要对表或索引进行 analyze。
+* `table_ids`：语句涉及到的表的 ID。
+* `txn_start_ts`：事务的开始时间戳，也是事务的唯一 ID，可以用这个值在 TiDB 日志中查找事务相关的其他日志。
+* `sql`：SQL 语句。
+
+和内存使用相关的字段：
+
+* `mem_max`：日志打印时语句已经使用的内存空间。该项使用两种单位标识内存使用量，分别为 Bytes 以及易于阅读的自适应单位。
```
Comment (yikeke) [inline dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md]:
```suggestion
* `mem_max`：日志打印时语句已经使用的内存空间。该项使用两种单位标识内存使用量，分别为 Bytes 以及易于阅读的自适应单位（比如 MB、GB 等）。
```
建议补充一两个自适应单位的例子，方便读者快速理解。（求 review 下补充的是否准确）

File: dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+和 SQL 执行的用户相关的字段：
+
+* `user`：执行语句的用户名。
+* `conn_id`：用户的链接 ID，可以用类似 `con:60026` 的关键字在 TiDB 日志中查找该链接相关的其他日志。
```
Comment (yikeke) [inline dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md]:
```suggestion
* `conn_id`：用户的连接 ID，可以用类似 `con:60026` 的关键字在 TiDB 日志中查找该连接相关的其他日志。
```
“连接”和“链接”哪个更准确？connection对应的应该是前者？

File: dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+* `database`：执行语句时使用的 database。
+
+和 TiKV Coprocessor Task 相关的字段：
+
+* `wait_time`：该语句在 TiKV 的等待时间之和，因为 TiKV 的 Coprocessor 线程数是有限的，当所有的 Coprocessor 线程都在工作的时候，请求会排队；当队列中有某些请求耗时很长的时候，后面的请求的等待时间都会增加。
+* `request_count`：该语句发送的 Coprocessor 请求的数量。
+* `total_keys`：Coprocessor 扫过的 key 的数量。
+* `processed_keys`：Coprocessor 处理的 key 的数量。相比 total_keys，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
```
Comment (yikeke) [inline dev/how-to/maintain/identify-abnormal-queries/identify-aborted-queries.md]:
```suggestion
* `processed_keys`：Coprocessor 处理的 key 的数量。与 total_keys 相比，processed_keys 不包含 MVCC 的旧版本。如果 processed_keys 和 total_keys 相差很大，说明旧版本比较多。
```

**Comment by XuHuaiyu** [comment]
PTAL @yikeke

**Comment by XuHuaiyu** [comment]
PTAL @yikeke

File: dev/how-to/maintain/identify-abnormal-queries/identify-slow-queries.md [hunk 6]
```diff
@@ -1,7 +1,7 @@
 ---
 title: 慢查询日志
 category: how-to
-aliases: ['/docs-cn/sql/slow-query/']
+aliases: ['/docs-cn/sql/slow-query/','/docs-cn/dev/how-to/maintain/identify-slow-queries/']
```
Comment (yikeke) [inline dev/how-to/maintain/identify-abnormal-queries/identify-slow-queries.md]: Please align this change in https://github.com/pingcap/docs/pull/1804/ @toutdesuite

Comment (toutdesuite) [reply dev/how-to/maintain/identify-abnormal-queries/identify-slow-queries.md]: Files are updated. PTAL. @yikeke

**Pull request merged**